src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java

```diff
@@ -585,4 +616,82 @@ private String mapToType(Reason reason) throws CTPException {
         throw new CTPException(Fault.SYSTEM_ERROR, "Unexpected refusal reason: %s", reason);
     }
   }
+
+  /**
+   * Make Case Service request to return cases by UPRN
+   *
+   * @param uprn of requested cases
+   * @param listCaseEvents boolean of whether require case events
+   * @return List of cases for UPRN
+   * @throws CTPException
+   */
+  private List<CaseDTO> callCaseSvcByUPRN(Long uprn, Boolean listCaseEvents) throws CTPException {
+
+    List<CaseContainerDTO> rmCases = new ArrayList<>();
+    try {
+      rmCases = caseServiceClient.getCaseByUprn(uprn, listCaseEvents);
+    } catch (ResponseStatusException ex) {
+      if (ex.getStatus() == HttpStatus.NOT_FOUND) {
+        log.with(uprn).info("Case by UPRN Not Found calling Case Service");
+        return Collections.emptyList();
+      } else {
+        log.with(uprn).with("status", ex.getStatus()).error("Error calling Case Service");
+        throw ex;
+      }
+    }
+
+    // Only return cases that are not of caseType = HI
+    List<CaseContainerDTO> casesToReturn =
+        (List<CaseContainerDTO>)
+            rmCases
+                .parallelStream()
+                .filter(c -> !(c.getCaseType().equals(CaseType.HI.name())))
+                .collect(Collectors.toList());
+
+    // Convert from Case service to Contact Centre DTOs
+    List<CaseDTO> caseServiceResponse = mapCaseContainerDTOList(casesToReturn);
+
+    // Clean up the events before returning them
+    caseServiceResponse.stream().forEach(c -> filterCaseEvents(c, listCaseEvents));
+    return caseServiceResponse;
+  }
+
+  /**
+   * Create new case, publish new address reported event and store new case in rerpository cache for
+   * new cases.
+   *
+   * @param uprn of address found
+   * @return Cachedcase details
+   * @throws CTPException
+   */
+  private CachedCase createNewCase(Long uprn) throws CTPException {
+
+    // Query AIMS for UPRN
+    Optional<AddressIndexAddressSplitDTO> addressQuery = addressSvc.uprnQuery(uprn);
+    if (addressQuery.isEmpty()) {
+      log.with("uprn", uprn).error("Failed to find address for UPRN");
```

Comment:
Will we ever get here? The call to uprnQuery() will throw an exception if a problem occurred, otherwise presumably we will have a valid address. In that case I would remove the optional, and code it as though if the method returns the address is good.

Reply:
Just wanted to carry on if null is returned. Agree is not likely to happen but is no overhead and wanted to program defensively. Will change if feel strongly but quite like it as is?

Reply:
Remove unnecessary code. Please change.

Reply:
Done.